lib/puppet/resource/type.rb

```diff
@@ -360,9 +360,6 @@ def set_resource_parameters(resource, scope)
     resource.add_parameters_from_consume
     inject_external_parameters(resource, scope)
 
-    resource_hash = {}
```

Comment:
You could say:
```
This method is known to only be used on the server/compile side.
```
I don't think that can be read to imply anything about any other method.